File: java/2068_Check_Whether_Two_Strings_are_Almost_Equivalent/2068_Check_Whether_Two_Strings_are_Almost_Equivalent.java
class Solution {
    public boolean checkAlmostEquivalent(String word1, String word2) {
        int[] letters = new int[26];
        for (char c : word1.toCharArray()) {
            letters[c-'a'] += 1;
        }
        for (char c : word2.toCharArray()) {
            letters[c-'a'] -= 1;
        }
        
        for (int i : letters) {
            if (Math.abs(i) > 3) {
                return false;
            }
        }
        return true;
    }
}

File: go/2068_Check_Whether_Two_Strings_are_Almost_Equivalent/2068_Check_Whether_Two_Strings_are_Almost_Equivalent.go
func checkAlmostEquivalent(word1 string, word2 string) bool {
    letters := make([]int, 26)
    for _, c := range word1 {
        letters[c-'a'] += 1
    }

    for _, c := range word2 {
        letters[c-'a'] -= 1
    }

    for _, c := range letters {
        if abs(c) > 3 {
            return false
        }
    }
    return true
}

func abs(a int) int {
    if a < 0 {
        return -1 * a
    } else {
        return a
    }
}
